### PR TITLE
Pca implement

### LINKED ISSUE #6: Perform clustering analysis on SONYC data

### PR DESCRIPTION
**Pull request by ChrisIck**
Demo notebook and training script that addresses and resolves #6

### CODE REVIEW

File: sonyc_kalman/train_pykalman.py [hunk 1]
```diff
@@ -50,7 +55,26 @@ def process_arguments(args):
...
+        pca_fit = PCA(n_components=params.n_pca)
+        pca_fit.fit(data)
+        data = pca_fit.transform(data)
+        
+        #save the PCA model
+        #dumps result to pickle file
+        if params.model_name is not None:
+            pca_name = params.model_name + '_pca.pkl'
+        else:
+            model_name = sensor_name + '_pca.pkl'
```
Comment (jtcramer) [inline sonyc_kalman/train_pykalman.py]: I think this should be `pca_name`?

File: sonyc_kalman/train_pykalman.py [hunk 2]
```diff
@@ -75,7 +99,7 @@ def process_arguments(args):
     #runs EM and stores result
     kf_trained = kf.em(data, n_iter=params.n_iter)
     
-    print('Training complete, saving result to {}'.format(params.output_path))
+    print('Training complete, saving result to {}'.format(os.path.join(params.output_path, model_name)))
```
Comment (jtcramer) [inline sonyc_kalman/train_pykalman.py]: This should probably go after `model_name` is defined below

Comment (jtcramer) [reply sonyc_kalman/train_pykalman.py]: Also the whole path should be defined once rather than recomputing it everytime

**Review by jtcramer — CHANGES_REQUESTED** [review]

File: sonyc_kalman/train_pykalman.py (hunk 1 above)
Comment (ChrisIck) [reply sonyc_kalman/train_pykalman.py]: Fixed in latest push

File: sonyc_kalman/train_pykalman.py (hunk 2 above)
Comment (ChrisIck) [reply sonyc_kalman/train_pykalman.py]: Fixed in latest push

**Pull request merged**